firstuseauthenticator/firstuseauthenticator.py

```diff
@@ -31,9 +31,11 @@ class CustomLoginHandler(LoginHandler):
     https://github.com/jupyterhub/firstuseauthenticator/pull/21#discussion_r364252009
     for more details
     """
-    custom_login_error = 'Invalid username or password'
+    custom_login_error = ''
     def _render(self, login_error=None, username=None):
-        return super()._render(self.custom_login_error, username)
+        if self.custom_login_error:
+            login_error = self.custom_login_error
+        return super()._render(login_error, username)
```

Comment:
The CustomLoginHandler is a class, and its instances with be used as LoginHandler instances. I'm wondering if these handlers map explicitly 1:1 with a new web requests, or 1:1 with a browser session with the tornado webserver somehow? If not, this will hide the error until it shows up by default for everyone.

Do you know how this works? When will a new LoginHandler be instanciated as an object from the class, is it once per web request?

Reply:
@consideRatio, I believe it's a handler per request. [LoginHandler is a BaseHandler](https://github.com/jupyterhub/jupyterhub/blob/f97ef7eaac73e4af4a5444b24fdffe47f388da70/jupyterhub/handlers/login.py#L89) and a [BaseHandler is a tornado RequestHandler](https://github.com/jupyterhub/jupyterhub/blob/2847c3a90c241b212b89aef6181b6da687948a03/jupyterhub/handlers/base.py#L67).
More about RequestHandler [here](https://www.tornadoweb.org/en/stable/guide/structure.html#subclassing-requesthandler).

Reply:
Ah! This was excellent documentation, and indeed it is 1:1 with the requests it seems from point 1 here https://www.tornadoweb.org/en/stable/guide/structure.html#overriding-requesthandler-methods